File: leetcode/easy/2027-minMovesToConvertString.java
// first pass solution
class Solution {
    public int minimumMoves(String s) {
        int i = 0;
        int moves = 0;
        
        while (i < s.length()) {
            if (s.charAt(i) == 'X') {
                moves++;
                i += 3;
            } else {
                i++;
            }
        }
        return moves;
    }
}

File: leetcode/easy/2027-minMovesToConvertString.js
var minimumMoves = function(s) {
    let i = 0;
    let moves = 0;
    
    while (i < s.length) {
        if (s[i] === 'X') {
            moves++;
            i += 2;
        }
        i++;
    }
    return moves;
};
